src/client/clientenvironment.cpp

```diff
@@ -392,7 +390,20 @@ void ClientEnvironment::addActiveObject(u16 id, u8 type,
 			<<std::endl;
 	}
 
-	addActiveObject(obj);
+	u16 new_id = addActiveObject(obj);
+
+	// Object initialized:
+	if ((obj = getActiveObject(new_id))) {
```

Comment:
Are the extra parenthesis needed?

Reply:
Technically no, but there will be a compiler warning if left out.